Source/SocketIO/Engine/SocketEngineSpec.swift

```diff
@@ -155,11 +155,16 @@ extension SocketEngineSpec {
         return com.url!
     }
 
-    func addHeaders(to req: inout URLRequest) {
-        if let cookies = cookies {
-            req.allHTTPHeaderFields = HTTPCookie.requestHeaderFields(with: cookies)
+    func addHeaders(to req: inout URLRequest, includingCookies additionalCookies: [HTTPCookie]? = nil) {
+        
+        var cookiesToAdd = [HTTPCookie]()
+        cookiesToAdd += cookies ?? []
```

Comment:
You can condense this line and the one above, right?

Reply:
yes, made this too